src/object.c

```diff
@@ -1812,3 +1803,89 @@ char *object_get_name(rct_object_entry *entry)
 
 	return (char *)pos;
 }
+
+void *object_aligned_alloc_pad(size_t alignment, size_t size)
+{
+	int remainder = size % alignment;
+	size_t zero_pad = (remainder > 0) ? (alignment - remainder) : 0;
+
+	// Allocate enough space for ...:
+	//  - the alignment
+	//  - a pointer to the original return value (needed for free)
+	//  - the object
+	//  - extra zero-padded space to bring object size to a multiple of 32
+	//  - extra space for alignment
+	void *ptr = (uint8 *)malloc(sizeof(size_t) + sizeof(uint8 *) + size + alignment - 1 + zero_pad);
+	if (ptr == NULL) {
+		log_error("Failed to allocate memory for object.");
+		assert(false);
+	}
+
+	// Align chunk pointer on boundary.
+	void *ptr_aligned = (void *)((uint8 *)ptr + sizeof(size_t) + sizeof(uint8 *) + alignment - 1 - (size % alignment));
+
+	// Save alignment for realloc
+	((size_t *)ptr_aligned)[-2] = alignment;
```

Comment:
- I am having trouble finding out what the mingw version of aligned malloc is (it might be `__mingw_aligned_malloc`?).
- It would require compiler macros for both `alloc` and `free` - Posix allows the pointer to be passed to free and the same goes for the other linux options, `_aligned_malloc` requires `_aligned_free`. I don't know about mingw, but possibly the same as MSVC.
- They all need different error handling. `posix_memalign` returns an error code, the others return a pointer or NULL (no `errno` set).
- This is the real kicker: The way initial object allocation works currently requires use of `realloc`. The versions that allow the pointer to be passed to free will work with `realloc`, but the aligned versions for mingw and msvc do not. I found it easier in the end to just implement the darn thing myself and save the trouble of having separate handling on all platforms.